File: QRClient/qrclient/src/main/java/com/yuan/qrclient/ui/activity/MainActivity.java
package com.yuan.qrclient.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;

import com.yuan.qrclient.R;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        findViewById(R.id.ac_scan).setOnClickListener(view ->
                startActivity(new Intent(MainActivity.this, ScanerActivity.class)));
        findViewById(R.id.ac_generate).setOnClickListener(view ->
                startActivity(new Intent(MainActivity.this, QRCodeActivity.class)));
    }

}

File: app/src/main/java/www/wiredcraft/testmobile/activity/MainActivity.kt
package www.wiredcraft.testmobile.activity

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import cn.bingoogolapple.refreshlayout.BGANormalRefreshViewHolder
import www.wiredcraft.testmobile.R
import www.wiredcraft.testmobile.api.MApi
import www.wiredcraft.testmobile.api.model.Item
import www.wiredcraft.testmobile.base.BaseActivity
import www.wiredcraft.testmobile.constant.MConstants
import www.wiredcraft.testmobile.databinding.ActivityMainBinding
import www.wiredcraft.testmobile.interfaces.MainModelInterface
import www.wiredcraft.testmobile.viewmodel.MainViewModel

/**
 *@Description:
 * # home
 * #0000      @Author: tianxiao     2022/3/30      onCreate
 */
class MainActivity : BaseActivity<ActivityMainBinding, MainViewModel>(), MainModelInterface {

    override fun createViewModel(): MainViewModel {
        return MainViewModel()
    }

    override fun onResume() {
        super.onResume()
        setStatusBarColor(R.color.white)
    }

    override fun layoutId(): Int {
        return R.layout.activity_main
    }

    override fun initialize(savedInstanceState: Bundle?) {
        binding.vm = viewModel
        binding.rv.layoutManager = LinearLayoutManager(this)
        binding.rv.adapter = viewModel.userAdapter
        binding.fresh.run {
            setDelegate(viewModel)
            setRefreshViewHolder(BGANormalRefreshViewHolder(this@MainActivity,
                true))
        }
        viewModel.userAdapter.interfaces = this
    }

    override fun StartActivityUserDetailsForResult(item: Item, position: Int) {
        startActivityLauncher.launch<UserDetailsActivity>(
            MConstants.KEY_Data to MApi.INITIALIZATION.gson.toJson(item)) { resultCode, _ ->
            if (resultCode == RESULT_OK) {
                // 处理回调结果
                viewModel.userAdapter.notifyItemChanged(position)
            }
        }
    }


}